cli/azd/pkg/project/scaffold_gen_environment_variables.go

```diff
@@ -556,6 +556,34 @@ func getResourceConnectionEnvs(usedResource *ResourceConfig,
 		default:
 			return []scaffold.Env{}, unsupportedAuthTypeError(resourceType, authType)
 		}
+	case ResourceTypeJavaEurekaServer:
+		return []scaffold.Env{
+			{
+				Name:  "eureka.client.register-with-eureka",
+				Value: "true",
+			},
+			{
+				Name:  "eureka.client.fetch-registry",
+				Value: "true",
+			},
+			{
+				Name:  "eureka.instance.prefer-ip-address",
+				Value: "true",
+			},
+			{
+				// Use '\${}' to escape parsing ${} placeholder in Bicep, just treat it as an Env of application
```

Comment:
Could you please move bicep related logic into `ToBicepEnv` in `bicep_env.go`? Because by design, this file should have nothing todo with bicep. And add unit test for it.

Reply:
Move this escape logic to `ServiceConfigEnv to Scaffold Env`

Reply:
Per teams sync, updated to use the value of `BASE_URL` directly, so that we can avoid escape issue.